File: scrapper.py
import requests
from bs4 import BeautifulSoup

# Take user input for URL
url = input("Enter the URL: ")

# Make a request to the website
res = requests.get(url)

# Use BeautifulSoup to parse the HTML
soup = BeautifulSoup(res.text, 'html.parser')

# Initialize empty lists for each file extension
php_links = []
js_links = []
html_links = []
css_links = []
image_links = []
audio_links = []
video_links = []

# Find all the links on the page
for link in soup.find_all('a'):
    # Get the href attribute of the link
    href = link.get('href')
    if href is not None:
        # Check the file extension and add the link to the appropriate list
        if href.endswith('.php'):
            php_links.append(href)
        elif href.endswith('.js'):
            js_links.append(href)
        elif href.endswith('.html'):
            html_links.append(href)
        elif href.endswith('.css'):
            css_links.append(href)
        elif href.endswith('.jpg') or href.endswith('.png') or href.endswith('.gif'):
            image_links.append(href)
        elif href.endswith('.mp3') or href.endswith('.wav'):
            audio_links.append(href)
        elif href.endswith('.mp4') or href.endswith('.avi'):
            video_links.append(href)

# Write the links to a text file
with open('links.txt', 'w') as file:
    file.write("PHP:\n")
    for link in php_links:
        file.write(link + '\n')
    file.write("\nJS:\n")
    for link in js_links:
        file.write(link + '\n')
    file.write("\nHTML:\n")
    for link in html_links:
        file.write(link + '\n')
    file.write("\nCSS:\n")
    for link in css_links:
        file.write(link + '\n')
    file.write("\nImages:\n")
    for link in image_links:
        file.write(link + '\n')
    file.write("\nAudio:\n")
    for link in audio_links:
        file.write(link + '\n')
    file.write("\nVideos:\n")
    for link in video_links:
        file.write(link + '\n')

File: scrapper.go
package main

import (
	"fmt"
	"io/ioutil"
	"net/http"
	"regexp"
	"strings"
)

func main() {
	var urls []string
	var url string

	fmt.Print("Enter the URL : ")
	fmt.Scan(&url)

	response, err := http.Get(url)
	if err != nil {
		fmt.Println(err)
		return
	}
	defer response.Body.Close()

	body, err := ioutil.ReadAll(response.Body)
	if err != nil {
		fmt.Println(err)
		return
	}

	// Find all URLs in the HTML
	re := regexp.MustCompile(`(https?|ftp|file)://[-A-Za-z0-9\+&@#/%?=~_|!:,.;]*[-A-Za-z0-9\+&@#/%=~_|]`)
	matches := re.FindAllString(string(body), -1)

	// Separate URLs by file type
	for _, match := range matches {
		if strings.HasSuffix(match, ".php") {
			urls = append(urls, "PHP:\n"+match)
		} else if strings.HasSuffix(match, ".js") {
			urls = append(urls, "JS:\n"+match)
		} else if strings.HasSuffix(match, ".html") {
			urls = append(urls, "HTML:\n"+match)
		} else if strings.HasSuffix(match, ".css") {
			urls = append(urls, "CSS:\n"+match)
		} else if strings.HasSuffix(match, ".jpg") || strings.HasSuffix(match, ".png") || strings.HasSuffix(match, ".gif") {
			urls = append(urls, "Image:\n"+match)
		} else if strings.HasSuffix(match, ".mp3") || strings.HasSuffix(match, ".wav") || strings.HasSuffix(match, ".ogg") {
			urls = append(urls, "Audio:\n"+match)
		} else if strings.HasSuffix(match, ".mp4") || strings.HasSuffix(match, ".avi") || strings.HasSuffix(match, ".mov") {
			urls = append(urls, "Video:\n"+match)
		}
	}

	// Write URLs to a text file
	err = ioutil.WriteFile("urls.txt", []byte(strings.Join(urls, "\n")), 0644)
	if err != nil {
		fmt.Println(err)
		return
	}

	fmt.Println("URLs have been scraped and written to 'urls.txt'.")
}
